typing/typemod.ml

```diff
@@ -1766,17 +1906,39 @@ let type_package env m p nl =
   (wrap_constraint env modl mty Tmodtype_implicit, tl')
 
 (* Fill in the forward declarations *)
+
+let type_open ?used_slot ovf env loc me =
+  type_open_ ?used_slot ?toplevel:None ovf env loc me
+
 let () =
   Typecore.type_module := type_module_alias;
   Typetexp.transl_modtype_longident := transl_modtype_longident;
   Typetexp.transl_modtype := transl_modtype;
-  Typecore.type_open := type_open_ ?toplevel:None;
+  Typecore.type_open := type_open;
   Typecore.type_package := type_package;
+  type_module_fwd := type_module;
   type_module_type_of_fwd := type_module_type_of
 
 
 (* Typecheck an implementation file *)
 
+let simplify_signature sg =
+  let rec aux = function
+    | [] -> [], StringSet.empty
+    | (Sig_value(id, _descr) as component) :: sg ->
+        let (sg, val_names) as k = aux sg in
+        let name = Ident.name id in
+        if StringSet.mem name val_names then k
+        else (component :: sg, StringSet.add name val_names)
+    | Sig_module({Ident.name}, _, _) :: sg
+        when String.contains name '#' -> aux sg
```

Comment:
Same remark as above: having a specific function rather `String.contains name '#'` would be nice.